db-init.sql

```diff
@@ -0,0 +1,166 @@
+	INSERT INTO PersonAttributeTypes VALUES (3, 'notes');
+
+-- PersonCategoryAttributes
+	-- what types of attributes a category of people have
+
+CREATE TABLE PersonCategoryAttributes (
+	personCategoryId SERIAL REFERENCES PersonCategories(personCategoryId) NOT NULL,
+	attrId SERIAL REFERENCES PersonAttributeTypes(attrId) NOT NULL
+);
+
+	-- farmer
+	INSERT INTO PersonCategoryAttributes VALUES (0, 2); -- paymentFrequency
+	INSERT INTO PersonCategoryAttributes VALUES (0, 3); -- notes
+	-- trader
+	INSERT INTO PersonCategoryAttributes VALUES (1, 0); -- username
+	INSERT INTO PersonCategoryAttributes VALUES (1, 1); -- password
+	-- admin
+	INSERT INTO PersonCategoryAttributes VALUES (2, 0); -- username
+	INSERT INTO PersonCategoryAttributes VALUES (2, 1); -- password
+	-- admin
+	INSERT INTO PersonCategoryAttributes VALUES (3, 0); -- username
+	INSERT INTO PersonCategoryAttributes VALUES (3, 1); -- password
+
+-- Person
+
+CREATE TABLE Persons (
+	personUuid UUID PRIMARY KEY NOT NULL DEFAULT uuid_generate_v4(),
+	firstName VARCHAR(255),
+	middleName VARCHAR(255),
+	lastName VARCHAR(255),
+	phoneNumber VARCHAR(20),
+	phoneArea VARCHAR(20),
+	phoneCountry VARCHAR(20),
+	companyName VARCHAR(255),
+	personCategoryId SERIAL REFERENCES PersonCategories(personCategoryId)
+);
+
+-- PersonAttributes
+	-- the values of the attributes that a specific person has
+
+CREATE TABLE PersonAttributes (
+	personUuid UUID REFERENCES Persons(personUuid) NOT NULL,
+	attrId SERIAL REFERENCES PersonAttributeTypes(attrId) NOT NULL,
+	attrValue VARCHAR(255)
+);
+
+-- MoneyTransactions
+
+CREATE TABLE MoneyTransactions (
+	moneyTransactionUuid UUID PRIMARY KEY NOT NULL DEFAULT uuid_generate_v4(),
+	datetime TIMESTAMP WITH TIME ZONE NOT NULL,
+	toPersonUuid UUID REFERENCES Persons(personUuid) NOT NULL,
+	fromPersonUuid UUID REFERENCES Persons(personUuid) NOT NULL,
+	amount MONEY NOT NULL
+);
+
+-- ProductTypes
+	-- all the different kinds of products
+
+CREATE TABLE ProductTypes (
+	productTypeId SERIAL PRIMARY KEY NOT NULL,
+	productName VARCHAR(255) NOT NULL,
+	productUnits VARCHAR(255) NOT NULL
+);
+
+	CREATE UNIQUE INDEX productNames_lower ON ProductTypes(lower(productName));
+
+	INSERT INTO ProductTypes VALUES (0, 'milk', 'litres');
+
+-- ProductTransactions
+
+CREATE TABLE ProductTransactions (
+	productTransactionUuid UUID PRIMARY KEY NOT NULL DEFAULT uuid_generate_v4(),
+	datetime TIMESTAMP WITH TIME ZONE NOT NULL,
+	toPersonUuid UUID REFERENCES Persons(personUuid) NOT NULL,
+	fromPersonUuid UUID REFERENCES Persons(personUuid) NOT NULL,
+	productTypeId SERIAL REFERENCES ProductTypes(productTypeId) NOT NULL,
+	amountOfProduct REAL NOT NULL,
+	costPerUnit MONEY NOT NULL
+);
+
+-- ProductTypeTransactionAttributes
+	-- the unique kinds of attributes a transaction for a type of product has
+	-- example: we track the milk density and milk temperature for all milk type transactions
+
+CREATE TABLE ProductTypeTransactionAttributes (
```

Comment:
what would we do if we need a new attribute for a product type that is composite? for example, we want a new attribute that is made up of 'value' and 'unit' attributes (like what we have been doing). would they just be 2 new attributes? 


I guess this is the case with milk density and I see that we will be combining the unit and value together in one field. My concern is if we want milk density in other units, we would have to make a new attribute for each available unit? 

also, does this mean that how our database is set up depends on the model of each company? and each company would get there own database? which I guess would make sense realistically, but might be a maintenance burden if we decide to add a new table or edit existing tables in all of the different databases. what do you think?

Reply:
For any kind of object, there should not be multiple columns for different units. All the units should be converted before being entered into the database.

This database could still work with multiple companies, I mean if we have this milk density attribute, there's no reason why a different company couldn't use the same attribute. It would be better to have a different database for each company though.

Reply:
ohh okay so it will be the client's responsibility to ensure that they convert it back to the system unit which i guess should make sense based on the name of the key

Reply:
or the core could convert it? what are your thoughts?

Reply:
yeah thinking about it more, I think the core should do all conversions if ever so that the logic is only in one place (instead of all clients)

Reply:
I think it's out of the scope of this PR. I think it shoudl be a client thing, so everything in our code we have the same units, right up until we show it to the user.